model-providers/watsonx/runtime/src/main/java/io/quarkiverse/langchain4j/watsonx/WatsonxStreamingChatModel.java

```diff
@@ -113,6 +98,111 @@ public void run() {
                         });
     }
 
+    @Override
+    public void generate(List<ChatMessage> messages, List<ToolSpecification> toolSpecifications,
+            StreamingResponseHandler<AiMessage> handler) {
+        TextGenerationRequest request = generateRequest(messages, toolSpecifications);
+        Context context = Context.of("response", new ArrayList<TextGenerationResponse>(), "toolExecution", false);
+
+        client.chatStreaming(request, version)
+                .subscribe()
+                .with(context,
+                        new Consumer<TextGenerationResponse>() {
+                            @Override
+                            @SuppressWarnings("unchecked")
+                            public void accept(TextGenerationResponse response) {
+                                try {
+
+                                    if (response == null || response.results() == null || response.results().isEmpty())
+                                        return;
```

Comment:
Is it a return or actually a completion signal?

Reply:
This is something that should never happen. This check could be removed.

Reply:
If we want to be precise, this should be a completion signal, but I don't think an empty list ever arrives from the

Reply:
I'm looking at the tests I've written, and this check is necessary in this scenario:

```
id: 1
event: message
data: {}

id: 2
event: message
data: {"modelId":"ibm/granite-13b-chat-v2","model_version":"2.1.0","created_at":"2024-05-04T14:29:19.162Z","results":[{"generated_text":"He","generated_token_count":0,"input_token_count":2,"stop_reason":"not_finished"}]}

...

id: N
event: close
data: {}
```

In this case, the first message doesn't return a `NullPointException`.

Reply:
Interesting, so it's not a completion signal.